service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedBigQueryDataTableResource.java

```diff
@@ -0,0 +1,216 @@
+      @JsonProperty("projectId") String projectId,
+      @JsonProperty("datasetName") String datasetName,
+      @JsonProperty("dataTableName") String dataTableName) {
+    super(workspaceId, resourceId, name, description, cloningInstructions);
+    this.projectId = projectId;
+    this.datasetName = datasetName;
+    this.dataTableName = dataTableName;
+    validate();
+  }
+
+  /**
+   * Constructor from database metadata
+   *
+   * @param dbResource database form of resources
+   */
+  public ReferencedBigQueryDataTableResource(DbResource dbResource) {
+    super(dbResource);
+    if (dbResource.getResourceType() != WsmResourceType.BIG_QUERY_DATATABLE) {
+      throw new InvalidMetadataException("Expected BIG_QUERY_DATATABLE");
+    }
+
+    ReferencedBigQueryDataTableAttributes attributes =
+        DbSerDes.fromJson(dbResource.getAttributes(), ReferencedBigQueryDataTableAttributes.class);
+    this.projectId = attributes.getProjectId();
+    this.datasetName = attributes.getDatasetName();
+    this.dataTableName = attributes.getDataTableName();
+    validate();
+  }
+
+  public static ReferencedBigQueryDataTableResource.Builder builder() {
+    return new ReferencedBigQueryDataTableResource.Builder();
+  }
+
+  public String getProjectId() {
+    return projectId;
+  }
+
+  public String getDatasetName() {
+    return datasetName;
+  }
+
+  public String getDataTableName() {
+    return dataTableName;
+  }
+
+  public ApiGcpBigQueryDataTableAttributes toApiAttributes() {
+    return new ApiGcpBigQueryDataTableAttributes()
+        .projectId(getProjectId())
+        .datasetId(getDatasetName())
+        .dataTableId(getDataTableName());
+  }
+
+  public ApiGcpBigQueryDataTableResource toApiResource() {
+    return new ApiGcpBigQueryDataTableResource()
+        .metadata(super.toApiMetadata())
+        .attributes(toApiAttributes());
+  }
+
+  @Override
+  public WsmResourceType getResourceType() {
+    return WsmResourceType.BIG_QUERY_DATATABLE;
+  }
+
+  @Override
+  public String attributesToJson() {
+    return DbSerDes.toJson(
+        new ReferencedBigQueryDataTableAttributes(
+            getProjectId(), getDatasetName(), getDataTableName()));
+  }
+
+  @Override
+  public void validate() {
+    super.validate();
+    if (Strings.isNullOrEmpty(getProjectId())
+        || Strings.isNullOrEmpty(getDatasetName())
+        || Strings.isNullOrEmpty(getDataTableName())) {
+      throw new MissingRequiredFieldException(
+          "Missing required field for ReferenceBigQueryDataTableAttributes");
+    }
+    ValidationUtils.validateBqDatasetName(getDatasetName());
+    ValidationUtils.validateBqDataTableName(getDataTableName());
+  }
+
+  @Override
+  public boolean checkAccess(FlightBeanBag context, AuthenticatedUserRequest userRequest) {
+    CrlService crlService = context.getCrlService();
+    return crlService.canReadBigQueryDataset(projectId, datasetName, userRequest);
```

Comment:
I believe there can be table-level access controls in BQ, so I'm not sure we can safely assume anyone with access to the dataset can access any table in it. https://cloud.google.com/bigquery/docs/table-access-controls-intro

Reply:
I added a separate check in crlService. It's pretty hacky... 

Also i'm getting a data table not found exception in the integration test.  do you know how to fix it? I think I need to create a bq data table but not sure where to do that.

Reply:
We do that a couple of places. One is [here](https://github.com/DataBiosphere/terra-workspace-manager/blob/4a8317959a3716868b3f374d65dc113555195f50/integration/src/main/java/scripts/testscripts/ControlledBigQueryDatasetLifecycle.java#L206), which makes a schema with  just one column (likely all you need). A more involved example is [here](https://github.com/DataBiosphere/terra-workspace-manager/blob/99289050f193bae0c9312ad45c1ec5728d27b0b4/integration/src/main/java/scripts/utils/ResourceModifier.java#L89).

Reply:
I ran into access permission exception when I tried to prgramatically create one using the createTable example but i ended up manually created on in the https://pantheon.corp.google.com/bigquery?project=terra-kernel-k8s.

Reply:
re: Pantheon, you may have better luck using our projects by starting from a chrome profile that has your Broad credentials and not your Verily/Google ones. Pantheon does things that break saved links and gives you links we can't share outside the company.

Not sure about the permission exception.

Reply:
oh sorry, this is the right link https://console.cloud.google.com/bigquery?project=terra-kernel-k8s&d=terra_wsm_test_dataset&p=terra-kernel-k8s&t=terra%20wsm%20test%20data%20table&page=table&ws=!1m5!1m4!4m3!1sterra-kernel-k8s!2sterra_wsm_test_dataset!3sterra%20wsm%20test%20data%20table